test/components/views/elements/MemberEventListSummary-test.js

```diff
@@ -20,42 +22,41 @@ describe.only('MemberEventListSummary', function() {
         });
     };
 
+    /**
+     * Generates a membership event with the target of the event set as a mocked RoomMember based
+     * on `parameters.userId`.
+     * @param {string} eventId the ID of the event.
+     * @param {object} parameters the parameters to use to create the event.
+     * @param {string} parameters.membership the membership to assign to `content.membership`
+     * @param {string} parameters.userId the state key and target userId of the event. If
+     * `parameters.senderId` is not specified, this is also used as the event sender.
+     * @param {string} parameters.prevMembership the membership to assign to
+     * `prev_content.membership`.
+     * @param {string} parameters.senderId the user ID of the sender of the event. Optional.
+     * Defaults to `parameters.userId`.
+     * @returns {MatrixEvent} the event created.
+     */
     const generateMembershipEvent = (eventId, parameters) => {
-        let membership = parameters.membership;
-        let userId = parameters.userId;
-        let prevMembership = parameters.prevMembership;
-        let senderId = parameters.senderId;
-        return {
-            content: {
-                membership: membership,
-            },
-            target: {
-                name: userId.match(/@([^:]*):/)[1],
+        let e = testUtils.mkMembership({
+            event: true,
+            user: parameters.senderId || parameters.userId,
+            skey: parameters.userId,
+            mship: parameters.membership,
+            prevMship: parameters.prevMembership,
+            target : {
+                name: parameters.userId.match(/@([^:]*):/)[1], // Use localpart as display name
+                userId: parameters.userId,
                 getAvatarUrl: () => {
                     return "avatar.jpeg";
                 },
-                userId: userId,
-            },
-            getId: () => {
-                return eventId;
-            },
-            getContent: function() {
-                return this.content;
-            },
-            getPrevContent: function() {
-                return {
-                    membership: prevMembership ? prevMembership : this.content,
-                };
             },
-            getSender: () => {
-                return senderId || userId;
-            },
-            getStateKey: () => {
-                return userId;
-            },
-        };
+        });
+        // Override random event ID
+        e.event.event_id = eventId;
```

Comment:
Because when the tests for expanded events are done, the shallow equality tests require a list of React instances (https://github.com/matrix-org/matrix-react-sdk/blob/bd106968ceafbad30766958008a6c840acc6e1aa/test/components/views/elements/MemberEventListSummary-test.js#L134) 

And the keys of these must match those generated in `generateTiles`, so they need to be constant, not random.